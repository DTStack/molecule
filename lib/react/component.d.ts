import { GlobalEvent } from 'mo/common/event';
export interface IComponent<S = any> {
    setState(values: S, callback?: (prevState: S, nextState: S) => void): void;
    render(nextState?: S): void;
    onUpdateState(callback: (prevState: S, nextState: S) => void): void;
    getState(): S;
    /**
     * Subscribe the component event
     * @param name
     * @param callback
     */
    onEvent(name: any, callback: any): void;
}
export declare abstract class Component<S = any> extends GlobalEvent implements IComponent<S> {
    protected abstract state: S;
    private _event;
    constructor();
    /**
     * Set the state values, and notify the view component to re render
     * @param values update target state values
     */
    setState(values: S, callback?: (prevState: S, nextState: S) => void): void;
    /**
     * Initiative notify the component to render the view by the state
     * @param nextState
     */
    render(nextState?: S): void;
    onUpdateState(callback: (prevState: S, nextState: S) => void): void;
    getState(): S;
    onEvent(name: any, callback: any): void;
}
