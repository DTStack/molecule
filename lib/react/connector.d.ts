import * as React from 'react';
import { IComponent } from './component';
import { Controller } from './controller';
export declare type ServiceObject = {
    [index: string]: IComponent;
};
export declare type ControllerObject = {
    [index: string]: Controller;
};
export declare function connect(Service: IComponent | ServiceObject, View: React.ComponentType<any>, Controller?: Controller | ControllerObject, watchFiled?: object): {
    new (props: any): {
        state: {
            lastUpdated: number;
        };
        componentDidMount(): void;
        onChange(prevState: any, nextState: any): void;
        update: () => void;
        getServiceState(): {};
        render(): JSX.Element;
        context: any;
        setState<K extends never>(state: {} | ((prevState: Readonly<{}>, props: Readonly<{}>) => {} | Pick<{}, K> | null) | Pick<{}, K> | null, callback?: (() => void) | undefined): void;
        forceUpdate(callback?: (() => void) | undefined): void;
        readonly props: Readonly<{}> & Readonly<{
            children?: React.ReactNode;
        }>;
        refs: {
            [key: string]: React.ReactInstance;
        };
        shouldComponentUpdate?(nextProps: Readonly<{}>, nextState: Readonly<{}>, nextContext: any): boolean;
        componentWillUnmount?(): void;
        componentDidCatch?(error: Error, errorInfo: React.ErrorInfo): void;
        getSnapshotBeforeUpdate?(prevProps: Readonly<{}>, prevState: Readonly<{}>): any;
        componentDidUpdate?(prevProps: Readonly<{}>, prevState: Readonly<{}>, snapshot?: any): void;
        componentWillMount?(): void;
        UNSAFE_componentWillMount?(): void;
        componentWillReceiveProps?(nextProps: Readonly<{}>, nextContext: any): void;
        UNSAFE_componentWillReceiveProps?(nextProps: Readonly<{}>, nextContext: any): void;
        componentWillUpdate?(nextProps: Readonly<{}>, nextState: Readonly<{}>, nextContext: any): void;
        UNSAFE_componentWillUpdate?(nextProps: Readonly<{}>, nextState: Readonly<{}>, nextContext: any): void;
    };
    contextType?: React.Context<any> | undefined;
};
