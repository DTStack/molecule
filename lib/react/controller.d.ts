import { GlobalEvent } from 'mo/common/event';
export declare abstract class Controller extends GlobalEvent {
}
