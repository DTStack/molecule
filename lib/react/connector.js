"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.connect = void 0;
var React = require("react");
var logger_1 = require("mo/common/logger");
function connect(Service, View, Controller, watchFiled) {
    return /** @class */ (function (_super) {
        __extends(Connector, _super);
        function Connector(props) {
            var _this = _super.call(this, props) || this;
            _this.update = function () {
                _this.setState({
                    lastUpdated: Date.now(),
                });
            };
            _this.onChange = _this.onChange.bind(_this);
            _this.state = {
                lastUpdated: Date.now(),
            };
            return _this;
        }
        Connector.prototype.componentDidMount = function () {
            if (Service.onUpdateState) {
                var service = Service;
                service.onUpdateState(this.onChange);
            }
            else {
                for (var name_1 in Service) {
                    if (name_1) {
                        var service = Service[name_1];
                        if (service.onUpdateState) {
                            service.onUpdateState(this.onChange);
                        }
                    }
                }
            }
        };
        Connector.prototype.onChange = function (prevState, nextState) {
            logger_1.default.info(prevState, nextState);
            if (!watchFiled) {
                this.update();
            }
            else {
                // TODO, 目前会全量触发更新，后期根据 watchField 字段来控制更新粒度
                // const prev = get(prevState, watchFiled);
                // const next = get(nextState, watchFiled);
                // if (!equals(prev, next)) {
                //     this.update();
                // }
            }
        };
        Connector.prototype.getServiceState = function () {
            var _a;
            var target = {};
            if (Service.onUpdateState) {
                var service = Service;
                Object.assign(target, __assign({}, service.getState()));
            }
            else {
                for (var name_2 in Service) {
                    if (name_2) {
                        var service = Service[name_2];
                        if (service.getState) {
                            Object.assign(target, (_a = {},
                                _a[name_2] = __assign({}, service.getState()),
                                _a));
                        }
                    }
                }
            }
            return target;
        };
        Connector.prototype.render = function () {
            return (React.createElement(View, __assign({}, this.state, this.getServiceState(), this.props, Controller)));
        };
        return Connector;
    }(React.Component));
}
exports.connect = connect;
//# sourceMappingURL=connector.js.map