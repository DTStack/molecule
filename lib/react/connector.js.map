{"version":3,"file":"connector.js","sourceRoot":"","sources":["../../src/react/connector.tsx"],"names":[],"mappings":";;;;;;;;;;;;;;;;;;;;;;;;;;;AAAA,6BAA+B;AAC/B,2CAAsC;AAYtC,SAAgB,OAAO,CACnB,OAAmC,EACnC,IAA8B,EAC9B,UAA0C,EAC1C,UAAmB;IAEnB;QAA+B,6BAAe;QAE1C,mBAAY,KAAK;YAAjB,YACI,kBAAM,KAAK,CAAC,SAKf;YAgCD,YAAM,GAAG;gBACL,KAAI,CAAC,QAAQ,CAAC;oBACV,WAAW,EAAE,IAAI,CAAC,GAAG,EAAE;iBAC1B,CAAC,CAAC;YACP,CAAC,CAAC;YAxCE,KAAI,CAAC,QAAQ,GAAG,KAAI,CAAC,QAAQ,CAAC,IAAI,CAAC,KAAI,CAAC,CAAC;YACzC,KAAI,CAAC,KAAK,GAAG;gBACT,WAAW,EAAE,IAAI,CAAC,GAAG,EAAE;aAC1B,CAAC;;QACN,CAAC;QAED,qCAAiB,GAAjB;YACI,IAAI,OAAO,CAAC,aAAa,EAAE;gBACvB,IAAM,OAAO,GAAG,OAAqB,CAAC;gBACtC,OAAO,CAAC,aAAa,CAAC,IAAI,CAAC,QAAQ,CAAC,CAAC;aACxC;iBAAM;gBACH,KAAK,IAAM,MAAI,IAAI,OAAO,EAAE;oBACxB,IAAI,MAAI,EAAE;wBACN,IAAM,OAAO,GAAe,OAAO,CAAC,MAAI,CAAC,CAAC;wBAC1C,IAAI,OAAO,CAAC,aAAa,EAAE;4BACvB,OAAO,CAAC,aAAa,CAAC,IAAI,CAAC,QAAQ,CAAC,CAAC;yBACxC;qBACJ;iBACJ;aACJ;QACL,CAAC;QAED,4BAAQ,GAAR,UAAS,SAAS,EAAE,SAAS;YACzB,gBAAM,CAAC,IAAI,CAAC,SAAS,EAAE,SAAS,CAAC,CAAC;YAClC,IAAI,CAAC,UAAU,EAAE;gBACb,IAAI,CAAC,MAAM,EAAE,CAAC;aACjB;iBAAM;gBACH,4CAA4C;gBAC5C,2CAA2C;gBAC3C,2CAA2C;gBAC3C,6BAA6B;gBAC7B,qBAAqB;gBACrB,IAAI;aACP;QACL,CAAC;QAQD,mCAAe,GAAf;;YACI,IAAM,MAAM,GAAG,EAAE,CAAC;YAClB,IAAI,OAAO,CAAC,aAAa,EAAE;gBACvB,IAAM,OAAO,GAAG,OAAqB,CAAC;gBACtC,MAAM,CAAC,MAAM,CAAC,MAAM,eAAO,OAAO,CAAC,QAAQ,EAAE,EAAG,CAAC;aACpD;iBAAM;gBACH,KAAK,IAAM,MAAI,IAAI,OAAO,EAAE;oBACxB,IAAI,MAAI,EAAE;wBACN,IAAM,OAAO,GAAe,OAAO,CAAC,MAAI,CAAC,CAAC;wBAC1C,IAAI,OAAO,CAAC,QAAQ,EAAE;4BAClB,MAAM,CAAC,MAAM,CAAC,MAAM;gCAChB,GAAC,MAAI,iBAAQ,OAAO,CAAC,QAAQ,EAAE,CAAE;oCACnC,CAAC;yBACN;qBACJ;iBACJ;aACJ;YACD,OAAO,MAAM,CAAC;QAClB,CAAC;QAED,0BAAM,GAAN;YACI,OAAO,CACH,oBAAC,IAAI,eACG,IAAI,CAAC,KAAK,EACV,IAAI,CAAC,eAAe,EAAE,EACtB,IAAI,CAAC,KAAK,EACV,UAAU,EAChB,CACL,CAAC;QACN,CAAC;QACL,gBAAC;IAAD,CAAC,AA5EM,CAAwB,KAAK,CAAC,SAAS,GA4E5C;AACN,CAAC;AAnFD,0BAmFC"}