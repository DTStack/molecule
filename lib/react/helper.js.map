{"version":3,"file":"helper.js","sourceRoot":"","sources":["../../src/react/helper.ts"],"names":[],"mappings":";;;AAAA,+BAA+D;AAE/D;;;;GAIG;AACH,SAAgB,kBAAkB,CAC9B,QAAyB,EACzB,KAAQ;IAER,OAAO,gBAAQ,CAAC,GAAG,CAAC,QAAQ,EAAE,UAAC,KAAK;QAChC,IAAI,sBAAc,CAAC,KAAK,CAAC,EAAE;YACvB,OAAO,oBAAY,CAAC,KAAK,EAAE,KAAK,CAAC,CAAC;SACrC;QACD,OAAO,KAAK,CAAC;IACjB,CAAC,CAAC,CAAC;AACP,CAAC;AAVD,gDAUC"}