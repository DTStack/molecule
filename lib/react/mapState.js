"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.mapState = void 0;
var React = require("react");
var logger_1 = require("mo/common/logger");
/**
 * Mapping the state to the component
 * TODO support mapping service method to the component.
 * @param WrappedComponent The component will be wrapped
 * @param state The state you want to injected, notice the state must be an observable object
 * @param subscribes The events of your subscribe, it used to trigger the state re render
 */
function mapState(WrappedComponent, state, actions) {
    return /** @class */ (function (_super) {
        __extends(StateProvider, _super);
        function StateProvider(props) {
            var _this = _super.call(this, props) || this;
            _this.onChange = _this.onChange.bind(_this);
            _this.state = {
                lastUpdated: Date.now(),
            };
            return _this;
        }
        StateProvider.prototype.componentDidMount = function () {
            // There is no declare state parameter as IObservable type, so must convert to any type.
            if (state.subscribe) {
                state.subscribe(this.onChange);
            }
            else {
                logger_1.default.error('The state parameter of mapState must be an observable object.');
            }
        };
        /**
         * TODO: Performance optimize, now whatever any properties changed in target,
         * there always be trigger the onChange event, so need a compare operation.
         * @param nextState changed data
         */
        StateProvider.prototype.onChange = function (nextState) {
            logger_1.default.info(nextState, state);
            this.setState({
                lastUpdated: Date.now(),
            });
        };
        StateProvider.prototype.render = function () {
            return (React.createElement(WrappedComponent, __assign({}, this.state, state, this.props, actions)));
        };
        return StateProvider;
    }(React.Component));
}
exports.mapState = mapState;
//# sourceMappingURL=mapState.js.map