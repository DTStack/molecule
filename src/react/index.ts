export * from './component';
export * from './helper';
export * from './connector';
export * from './controller';
