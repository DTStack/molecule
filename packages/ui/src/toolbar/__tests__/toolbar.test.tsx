import React from 'react';
import { cleanup, render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { Toolbar, toolbarClassName } from '..';

const toolbarData = [
    {
        id: 'action1',
        title: 'mock action title',
    },
    {
        id: 'action2',
        title: 'mock action title',
    },
];

afterEach(cleanup);

describe('The Toolbar Component', () => {
    test('Match Snapshot', () => {
        const component = renderer.create(<Toolbar data={toolbarData} />);
        const toolbar = component.toJSON();
        expect(toolbar).toMatchSnapshot();
    });

    test('should render extra classNames', () => {
        const testClassNames = 'test-classNames';
        const { container } = render(
            <Toolbar className={testClassNames} data={toolbarData} />
        );
        const toolbar = container.querySelector(`.${toolbarClassName}`);
        expect(toolbar?.classList).toContain(testClassNames);
    });

    test('Should pass throught params to ActionBar', async () => {
        const testId = 'test-toolbar';
        const { findByTestId } = render(
            <Toolbar data-testid={testId} data={toolbarData} />
        );
        const toolbar = await findByTestId(testId);
        expect(toolbar.parentElement?.classList).toContain(toolbarClassName);
    });
});
