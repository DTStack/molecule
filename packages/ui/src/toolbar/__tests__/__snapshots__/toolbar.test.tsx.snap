// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Toolbar Component Match Snapshot 1`] = `
<div
  className="mo-tool-bar"
>
  <div
    className="mo-action-bar"
  >
    <ul
      className="mo-action-bar__container"
    >
      <li
        className="mo-action-bar__item"
        data-index={0}
        onClick={[Function]}
      />
      <li
        className="mo-action-bar__item"
        data-index={1}
        onClick={[Function]}
      />
    </ul>
  </div>
</div>
`;
